2-mandatory/3-groceries.js

```diff
@@ -27,13 +27,24 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
-
+for (let key in weeklyMealPlan) {
+  weeklyMealPlan[key].forEach((food) => {
+    if (weeklyGroceriesToBuy.indexOf(food) === -1) {
+      weeklyGroceriesToBuy.push(food);
+    }
+  });
+}
 /*
 Exercise 2:
   Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
 */
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
+for (let key in weeklyMealPlan) {
+  if (key === "saturday" || key === "sunday") {
+    weeklyMealPlan[key].forEach((food) => weekendGroceriesToBuy.push(food));
```

Comment:
You should also include a check here to make sure the item is not already in the list. In the example you have been given, there are no repeating items for Saturday and Sunday so your tests still pass, but your code should be able to deal with different inputs.